### ISSUE TITLE
No longer able to target older macOS

### ISSUE THREAD

**Issue by wagenet** [rust-lang/rust#40481]
I'm using `MACOSX_DEPLOYMENT_TARGET=10.7` and `-Wl,-macosx_version_min,10.7` in my Rust application. In current nightlies this no longer works and I see a bunch of messages like:

```
ld: warning: object file (target/dist/x86_64-darwin/objs/absvdi2.o) was built for newer OSX version (10.12) than being linked (10.7)
```

and

```
ld: warning: linker symbol '$ld$hide$os10.4$___absvdi2' hides a non-existent symbol '___absvdi2'
```

The last good nightly was 2017-02-13. It was suggested that this may be due to a shift to a build system.

**Comment by alexcrichton** [comment]
This was also flagged by Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1337955
